### PR TITLE
Move ImuData to sensor. Add ImuData proto

### PR DESCRIPTION
**Pull request by jihoonl**
Related to #253

### CODE REVIEW

File: cartographer/mapping_3d/imu_integration.h [hunk 1]
```diff
@@ -24,17 +24,12 @@
 #include "Eigen/Geometry"
 #include "cartographer/common/time.h"
 #include "cartographer/transform/transform.h"
+#include "cartographer/sensor/imu_data.h"
```
Comment (wohe) [inline cartographer/mapping_3d/imu_integration.h]: This should be sorted alphabetically. Did you run `clang-format`?

File: cartographer/sensor/imu_data.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+// Converts 'imu_data' to a proto::ImuData.
+proto::ImuData ToProto(const ImuData& imu_data);
+
+// Converts 'proto' to a ImuData.
+ImuData FromProto(const proto::ImuData& proto);
+
+} // namespace sensor
+} // namespace cartographer
+
+#endif
```
Comment (wohe) [inline cartographer/sensor/imu_data.h]: Style:
```
#endif  // CARTOGRAPHER_SENSOR_IMU_DATA_H_
```

File: cartographer/sensor/proto/sensor.proto [hunk 3]
```diff
@@ -40,3 +40,10 @@ message RangeData {
   optional PointCloud point_cloud = 2;
   optional PointCloud missing_echo_point_cloud = 3;
 }
+
+// Proto representation of ::cartographer::sensor::ImuData
+message ImuData {
+  optional int64 timestamp = 1;
+  optional transform.proto.Vector3d linear_acceleration = 2;
+  optional transform.proto.Vector3d angular_velocity= 3;
```
Comment (wohe) [inline cartographer/sensor/proto/sensor.proto]: Style: Space before `=`.

File: cartographer/sensor/imu_data.h [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+#ifndef CARTOGRAPHER_SENSOR_IMU_DATA_H_
+#define CARTOGRAPHER_SENSOR_IMU_DATA_H_
+
+#include "Eigen/Core"
+#include "cartographer/common/time.h"
+#include "cartographer/sensor/proto/sensor.pb.h"
+
+namespace cartographer {
+namespace sensor{
```
Comment (wohe) [inline cartographer/sensor/imu_data.h]: Style: Space before '{'.

File: cartographer/sensor/imu_data.h [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+struct ImuData {
+  common::Time time;
+  Eigen::Vector3d linear_acceleration;
+  Eigen::Vector3d angular_velocity;
+};
+
+// Converts 'imu_data' to a proto::ImuData.
+proto::ImuData ToProto(const ImuData& imu_data);
+
+// Converts 'proto' to a ImuData.
```
Comment (wohe) [inline cartographer/sensor/imu_data.h]: Either `to ImuData` or `to an ImuData`.

File: cartographer/sensor/imu_data.cc [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  return proto;
+}
+
+ImuData FromProto(const proto::ImuData& proto) {
+  auto imu_data = ImuData{
+      common::FromUniversal(proto.timestamp()),
+      transform::ToEigen(proto.linear_acceleration()),
+      transform::ToEigen(proto.angular_velocity())
+  };
+  return imu_data;
```
Comment (wohe) [inline cartographer/sensor/imu_data.cc]: Not need for the local variable, i.e. `return ImuData{...`?

**Comment by jihoonl** [comment]
Fixed.

File: cartographer/mapping_3d/sparse_pose_graph/optimization_problem.cc [hunk 7]
```diff
@@ -32,6 +32,7 @@
 #include "cartographer/mapping_3d/acceleration_cost_function.h"
 #include "cartographer/mapping_3d/ceres_pose.h"
 #include "cartographer/mapping_3d/rotation_cost_function.h"
+#include "cartographer/mapping_3d/imu_integration.h"
```
Comment (wohe) [inline cartographer/mapping_3d/sparse_pose_graph/optimization_problem.cc]: Please sort alphabetically. This belongs one line up. Is `clang-format` not doing this for you?

Comment (jihoonl) [inline cartographer/mapping_3d/sparse_pose_graph/optimization_problem.cc]: Ah sorry. I thought I ran for all files.

**Comment by jihoonl** [comment]
ok. All formatting issue should have been fixed.

**Pull request merged**